src/test/java/com/cloudant/tests/ViewPaginationTests.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (c) 2015 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.tests;
+
+import static org.junit.Assert.assertEquals;
+
+import com.cloudant.client.api.CloudantClient;
+import com.cloudant.client.api.Database;
+import com.cloudant.client.api.views.Key;
+import com.cloudant.client.api.views.ViewResponse;
+import com.cloudant.tests.util.CheckPagination;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+
+@RunWith(Parameterized.class)
+public class ViewPaginationTests {
+
+    /**
+     * Parameters for these tests so we run each test multiple times.
+     * We run with a single key or a complex key and both ascending and descending.
+     */
+    @Parameterized.Parameters(name = "Key:{0},Descending:{1}")
+    public static Iterable<Object[]> data() {
+        return Arrays.asList(new Object[][]{
+                {CheckPagination.Type.SINGLE, true},
+                {CheckPagination.Type.SINGLE, false},
+                {CheckPagination.Type.COMPLEX, false},
+                {CheckPagination.Type.COMPLEX, true},
+        });
+    }
+
+    @Parameterized.Parameter
+    public CheckPagination.Type type;
+
+    @Parameterized.Parameter(value = 1)
+    public boolean descending;
+
+    private static Database db;
+    private CloudantClient account;
+
+    @Before
+    public void setUp() {
+        account = CloudantClientHelper.getClient();
+
+        db = account.database("lightcouch-db-test", true);
```

Comment:
Yes, this was copied straight from the existing test cases and is a pattern used throughout java-cloudant. This is as good a place as any to start fixing it: c340d412c502fca8432d9af8c24c9a6fd18d43a2